cpp/modmesh/CMakeLists.txt

```diff
@@ -49,10 +49,16 @@ set(MODMESH_ROOT_HEADERS
     ${CMAKE_CURRENT_SOURCE_DIR}/base.hpp
     ${CMAKE_CURRENT_SOURCE_DIR}/math.hpp
     ${CMAKE_CURRENT_SOURCE_DIR}/grid.hpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/SerializableItem.hpp
```

Comment:
I would like the include files in the source root directory to be all lower-case because it looks nicer in my opinion.

It's purely my personal preference.  I am stubborn about it but open to discussions.

Reply:
I see. I moved the serialization to the `serialization` directory.

Reply:
Sounds good.